### PR TITLE
Stylistic improvements to notebooks/Training_a_LoRA_With_Instruct_Lab.ipynb

### PR DESCRIPTION
**Pull request by fc7**
# Changes

**Description of your changes:**

This PR brings some minor changes to the text of the Jupyter notebook `notebooks/Training_a_LoRA_With_Instruct_Lab.ipynb`, mainly to improve readability.

* Updates the name of the cli (`lab` -> `ilab`)
* Updates the name of the project (InstructLab, not `instruct lab`)
* Makes some stylistic improvements
* fixes a few typos

The content itself, including the code, is untouched.

Please ignore whitespace with `&w=1` when looking at the diff.

### CODE REVIEW

**Comment by fc7** [comment]
I closed this by mistake. Will reopen it after reapplying my initial commit.

**Comment by fc7** [comment]
@rawkintrevo regarding your suggestion above to add the colab badge, I would suggest adding it to `notebooks/README.md` at the beginning of the subsection `## Running the notebook`. It looks like this in my preview:
Screenshot 2024-05-10 at 08 53 26

Should I add this to this PR?

**Comment by rawkintrevo** [comment]
@fc7 that looks right yes. I want to continue to clarify, that I no longer have rights on this repo to merge (and am going out of pocket to hang out with wife and new baby any day now so won't even be seeing messages soon), but this looks good to me and someone else (maybe @hickeyma  or @xukai92 or their delegates) will have to take it from here.

**Comment by fc7** [comment]
With the last commit, I have followed @rawkintrevo's suggestion with the Colab badge, and made some additional minor improvements in that README with regard to style and grammar. I have also put the deprecated Kaggle instructions in a collapsed section, to ease readability.

**Comment by leseb** [comment]
Thanks for working on this, I'd like to move forward once we merge https://github.com/instructlab/instructlab/pull/1202 so we make sure the newly provided notebook is valid. Thanks for your patience!

**Comment by fc7** [comment]
Thanks @leseb 
In the meanwhile I just checked with your `scripts/validate_notebook.py` that the notebook in my fork also passes that check, just to be on the safe side.

**Comment by leseb** [comment]
> Thanks @leseb In the meanwhile I just checked with your `scripts/validate_notebook.py` that the notebook in my fork also passes that check, just to be on the safe side.

Great thanks! Also I just noticed that GitHub has a built-in UI validator for this: https://github.com/instructlab/instructlab/blob/35a4617ad862f5b78ce890bf36e2e424e43130fb/notebooks/Training_a_LoRA_With_Instruct_Lab.ipynb, if it renders then it's valid.

**Review by hickeyma — CHANGES_REQUESTED** [review]
@fc7 Can you please fix the CI failures? We should be good to go then.

**Comment by fc7** [comment]
@hickeyma the CI failures should be fixed now

**Comment by leseb** [comment]
@Mergifyio rebase

**Review by leseb — APPROVED** [review]
Thanks! The notebook linter works too :)

**Review by nathan-weinberg — DISMISSED** [review]

**Comment by leseb** [comment]
@fc7 please rebase :) so we can go ahead with this change. Thanks!

**Comment by leseb** [comment]
@hickeyma PTAL, ready for final review. Thanks!

**Comment by leseb** [comment]
@fc7 please rebase one last time 🙏🏻 , almost there, thanks!

**Review by hickeyma — COMMENTED** [review]
Thanks for the patience @fc7. Once rebased, lets get this merged.

**Comment by fc7** [comment]
@hickeyma rebase done

**Review by leseb — COMMENTED** [review]
Please squash all the commits so we can proceed with the merge, thanks!

**Review by hickeyma — DISMISSED** [review]
LGTM, thanks @fc7 for the updates and your patience with the changes.

File: notebooks/README.md
Comment (nathan-weinberg) [inline notebooks/README.md:40]:
```suggestion
1. Uploading the output of `ilab generate` (a synthetic dataset created based on your hand-written prompts/responses).
1. Checking the base model before training
1. Setting up and training a LoRA. A LoRA uses Parameter-Efficient Fine-Tuning (PEFT) methods to fine-tune a model on a small subset of the overall parameters, allowing you to conduct fine-tuning in a fraction of the time, on a fraction of the hardware required. The resultant model should be updated and better handle your queries than the base model.
1. Inspecting the output model to make sure the LoRA training had the desired effect (i.e. the output has improved).
```

Comment (nathan-weinberg) [reply notebooks/README.md:40]: These should be left as they were - this does automatic numbering in Markdown

File: .markdownlint-cli2.yaml
Comment (nathan-weinberg) [inline .markdownlint-cli2.yaml:12]: Why do we need to have inline HTML?

Comment (fc7) [reply .markdownlint-cli2.yaml:12]: Because of the Google Colab badge which I have added to `notebooks/README.md` following the suggestion by @rawkintrevo [here](https://github.com/instructlab/instructlab/pull/1053#issuecomment-2093132317)

File: notebooks/README.md
Comment (fc7) [reply notebooks/README.md:40]: Ok I will change this. I guess the Jupyter extension of VS Code changed this automatically.

Comment (fc7) [reply notebooks/README.md]: Done

Comment (nathan-weinberg) [reply notebooks/README.md]: Thanks!

File: .markdownlint-cli2.yaml
Comment (nathan-weinberg) [reply .markdownlint-cli2.yaml:12]: Ah I see, thanks for the explanation!

**Review by nathan-weinberg — DISMISSED** [review]

**Comment by fc7** [comment]
> Please squash all the commits so we can proceed with the merge, thanks!

@leseb Shouldn't this rather be done on your side during the merge? Github has [introduced this feature](https://github.blog/2016-04-01-squash-your-commits/) a long time ago.

**Comment by leseb** [comment]
> > Please squash all the commits so we can proceed with the merge, thanks!
> 
> @leseb Shouldn't this rather be done on your side during the merge? Github has [introduced this feature](https://github.blog/2016-04-01-squash-your-commits/) a long time ago.

I'm not the one merging, our bot Mergify does and it is not configured to squash commits. So please squash and push again. Thanks!

**Comment by nathan-weinberg** [comment]
@fc7 @leseb You can read here about the project merge policy: https://github.com/instructlab/dev-docs/blob/main/docs/github-merge-strategy.md#method-for-merging-a-pull-request

**Comment by fc7** [comment]
Thanks @nathan-weinberg, then based on the project merge policy, I will indeed refrain from squashing anything, and leave the commit history intact. @leseb can you please approve? 🙏

**Comment by nathan-weinberg** [comment]
@fc7 I wouldn't say your current series of commits aligns with this
```text
The PR itself should NOT include any merge commits of main back into the developer's branch. We expect the proposed commit series to be a clean set of commits against main without conflicts or merge commit history. 
```
Squashing locally is easy to do - if you need help with the steps we can assist you.

**Review by leseb — CHANGES_REQUESTED** [review]
Just to make it clear, I won't approve until:

* Commits are squashed into a single one, this one: "[Training notebook: Improve style, fix typos, update name of cli](https://github.com/instructlab/instructlab/pull/1053/commits/387d9d4c459ee3cc3f48599baed0ec6a59afda9b)"
* A sign-off is decided, currently the commit has two sign-off "Signed-off-by: fc7 <firmicus04@gmail.com>
Signed-off-by: François Charette <firmicus04@gmail.com>"

Thanks!

**Comment by fc7** [comment]
@nathan-weinberg @leseb Thanks for the clarification. I believe I have done the necessary steps locally but the interactive merge was not as trivial as I thought

**Comment by fc7** [comment]
@nathan-weinberg @leseb I have squashed all my commits into my initial one and used one signoff. Hopefully this PR is compliant now. And sorry for misunderstanding the merge policy earlier on.

**Comment by leseb** [comment]
@fc7 I think there is a problem with your last push. Can you just pick your commit and rebase on top of the current main? Thanks!

**Comment by nathan-weinberg** [comment]
Holding until commits are fixed

**Comment by fc7** [comment]
Apologies for the delay, I was travelling and did not pay attention. I also noticed the issue with the commits after my latest push. I will rebase asap.

**Comment by fc7** [comment]
@leseb and @nathan-weinberg: PR ready for review

**Review by leseb — APPROVED** [review]

**Review by nathan-weinberg — APPROVED** [review]

**Pull request merged**
